validations.go

```diff
@@ -30,6 +32,21 @@ type validateCreateable interface {
 }
 
 func (m *Model) validateCreate(c *Connection) (*validate.Errors, error) {
+	v := reflect.Indirect(reflect.ValueOf(m.Value))
```

Comment:
I feel like a lot of this logic is repeated. would make sense to break it out into it's own method?